src/pipdeptree/_models/dag.py

```diff
@@ -110,16 +106,8 @@ def filter_nodes(self, include: list[str] | None, exclude: set[str] | None) -> P
         if include is None and exclude is None:
             return self
 
-        # Note: In following comparisons, we use lower cased values so
-        # that user may specify `key` or `project_name`. As per the
-        # documentation, `key` is simply
-        # `project_name.lower()`. Refer:
-        # https://setuptools.readthedocs.io/en/latest/pkg_resources.html#distribution-objects
-        include_with_casing_preserved: list[str] = []
-        if include:
-            include_with_casing_preserved = include
-            include = [s.lower() for s in include]
-        exclude = {s.lower() for s in exclude} if exclude else set()
+        exclude = exclude if exclude else set()
+        include = include if include else []
```

Comment:
I don't think we should change this as elements in `include` and `exclude` comes straight from user input and we should normalize what they give us (but now that I see it, I think we should do better and perform PEP 503 normalization here on the strings).

Reply:
fixed this

Reply:
The reason why the `include_casing_preserved` array exists is so that when the user passes package names that don't match, the names will have their casing preserved when they are added to the error string that is printed out. By normalizing them, users would get "foobar" rather than the expected "FooBar". There maybe better ways to implement this, but seeing how complicated this PR is starting to get let's just leave this out for some other time.

Reply:
Oh, at first, I didn't consider this reason, and I was troubled as to why it had to be done this way. But now I understand, and I will keep this part.